### PR TITLE
Fix Tiling NNCF

### PR DESCRIPTION
**Pull request by eugene123tw**
### Summary
Fix Tiling NNCF test pipeline

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: otx/algorithms/detection/adapters/mmdet/nncf/builder.py [hunk 1]
```diff
@@ -132,7 +132,12 @@ def build_nncf_detector(  # pylint: disable=too-many-locals,too-many-statements
         )
         state_dict = None
 
-    test_pipeline = [LoadImage()] + config.data.test.pipeline[1:]
+    test_pipeline = [LoadImage()]
+    if len(config.data.test.pipeline) > 1:
+        test_pipeline.extend(config.data.test.pipeline[1:])
+    else:
+        test_pipeline.extend(config.data.test.pipeline)
+
```
Comment (cih9088) [inline otx/algorithms/detection/adapters/mmdet/nncf/builder.py]:
```suggestion
    # replace image loading pipeline with `LoadImage()`
    test_pipeline = [LoadImage()] + [
        pipeline for pipeline in config.data.test.pipeline if not pipeline.type.startswith("LoadImage")
    ]
```

**Review by jaegukhyun — CHANGES_REQUESTED** [review]
Overall implementation looks good to me. Could you check integration test failure, In addition it would be nice if you add e2e tiling det, iseg test results

File: otx/algorithms/detection/adapters/mmdet/nncf/builder.py (hunk 1 above)
Comment (eugene123tw) [reply otx/algorithms/detection/adapters/mmdet/nncf/builder.py]: Done

**Review by jaegukhyun — COMMENTED** [review]
Could you explain why we need separate dataset for tiling instance segmentation?

**Comment by eugene123tw** [comment]
> Could you explain why we need separate dataset for tiling instance segmentation?

The change is to ensure a more effective evaluation of our tiling capabilities. While the current dataset may be appropriate for normal object detection, it contains large objects that do not effectively test the limits of our tiling algorithms. Therefore, I think it'd be better to use a more suitable dataset to assess our tiling capabilities accurately.

**Review by cih9088 — DISMISSED** [review]

**Comment by eugene123tw** [comment]
@jaegukhyun I ran the e2e tests are there are 3 failed cases related to nncf validate fq

```
==================================================================================================================================== FAILURES ====================================================================================================================================
_______________________________________________________________________________________________ TestToolsTilingDetection.test_nncf_validate_fq[Custom_Object_Detection_Gen3_ATSS] ________________________________________________________________________________________________

self = <tests.e2e.cli.detection.test_tiling_detection.TestToolsTilingDetection object at 0x7f2884c33e80>
template = ModelTemplate(model_template_id='Custom_Object_Detection_Gen3_ATSS', model_template_path='/home/yuchunli/training_exte...paths=ExportableCodePaths(default=None, openvino=None), task_type_sort_priority=-1, gigaflops=20.6, size=9.1, hpo=None)
tmp_dir_path = PosixPath('/tmp/tmp278z3arm/tiling_det')

    @e2e_pytest_component
    @pytest.mark.skipif(TT_STABILITY_TESTS, reason="This is TT_STABILITY_TESTS")
    @pytest.mark.parametrize("template", templates, ids=templates_ids)
    def test_nncf_validate_fq(self, template, tmp_dir_path):
        tmp_dir_path = tmp_dir_path / "tiling_det"
        if template.entrypoints.nncf is None:
            pytest.skip("nncf entrypoint is none")
    
>       nncf_validate_fq_testing(template, tmp_dir_path, otx_dir, "detection", type(self).__name__)

tests/e2e/cli/detection/test_tiling_detection.py:219: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/test_suite/run_test_command.py:559: in nncf_validate_fq_testing
    _validate_fq_in_xml(xml_path, path_to_ref_data, "nncf", test_name)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

xml_path = '/tmp/tmp278z3arm/tiling_det/otx/algorithms/detection/configs/detection/mobilenetv2_atss/exported_nncf_Custom_Object_Detection_Gen3_ATSS/openvino.xml'
path_to_ref_data = '/home/yuchunli/training_extensions/tests/e2e/cli/detection/reference/Custom_Object_Detection_Gen3_ATSS/compressed_model.yml', compression_type = 'nncf', test_name = 'TestToolsTilingDetection'

    def _validate_fq_in_xml(xml_path, path_to_ref_data, compression_type, test_name):
        num_fq = get_number_of_fakequantizers_in_xml(xml_path)
        assert os.path.exists(path_to_ref_data), f"Reference file does not exist: {path_to_ref_data} [num_fq = {num_fq}]"
    
        with open(path_to_ref_data, encoding="utf-8") as stream:
            ref_data = yaml.safe_load(stream)
        ref_num_fq = ref_data.get(test_name, {}).get(compression_type, {}).get("number_of_fakequantizers", -1)
>       assert num_fq == ref_num_fq, f"Incorrect number of FQs in optimized model: {num_fq} != {ref_num_fq}"
E       AssertionError: Incorrect number of FQs in optimized model: 155 != -1

tests/test_suite/run_test_command.py:475: AssertionError
________________________________________________________________________________________________ TestToolsTilingDetection.test_nncf_validate_fq[Custom_Object_Detection_Gen3_SSD] ________________________________________________________________________________________________

self = <tests.e2e.cli.detection.test_tiling_detection.TestToolsTilingDetection object at 0x7f2884c321d0>
template = ModelTemplate(model_template_id='Custom_Object_Detection_Gen3_SSD', model_template_path='/home/yuchunli/training_exten..._paths=ExportableCodePaths(default=None, openvino=None), task_type_sort_priority=-1, gigaflops=9.4, size=7.6, hpo=None)
tmp_dir_path = PosixPath('/tmp/tmp278z3arm/tiling_det')

    @e2e_pytest_component
    @pytest.mark.skipif(TT_STABILITY_TESTS, reason="This is TT_STABILITY_TESTS")
    @pytest.mark.parametrize("template", templates, ids=templates_ids)
    def test_nncf_validate_fq(self, template, tmp_dir_path):
        tmp_dir_path = tmp_dir_path / "tiling_det"
        if template.entrypoints.nncf is None:
            pytest.skip("nncf entrypoint is none")
    
>       nncf_validate_fq_testing(template, tmp_dir_path, otx_dir, "detection", type(self).__name__)

tests/e2e/cli/detection/test_tiling_detection.py:219: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/test_suite/run_test_command.py:559: in nncf_validate_fq_testing
    _validate_fq_in_xml(xml_path, path_to_ref_data, "nncf", test_name)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

xml_path = '/tmp/tmp278z3arm/tiling_det/otx/algorithms/detection/configs/detection/mobilenetv2_ssd/exported_nncf_Custom_Object_Detection_Gen3_SSD/openvino.xml'
path_to_ref_data = '/home/yuchunli/training_extensions/tests/e2e/cli/detection/reference/Custom_Object_Detection_Gen3_SSD/compressed_model.yml', compression_type = 'nncf', test_name = 'TestToolsTilingDetection'

    def _validate_fq_in_xml(xml_path, path_to_ref_data, compression_type, test_name):
        num_fq = get_number_of_fakequantizers_in_xml(xml_path)
        assert os.path.exists(path_to_ref_data), f"Reference file does not exist: {path_to_ref_data} [num_fq = {num_fq}]"
    
        with open(path_to_ref_data, encoding="utf-8") as stream:
            ref_data = yaml.safe_load(stream)
        ref_num_fq = ref_data.get(test_name, {}).get(compression_type, {}).get("number_of_fakequantizers", -1)
>       assert num_fq == ref_num_fq, f"Incorrect number of FQs in optimized model: {num_fq} != {ref_num_fq}"
E       AssertionError: Incorrect number of FQs in optimized model: 67 != -1

tests/test_suite/run_test_command.py:475: AssertionError
_________________________________________________________________________________________________ TestToolsTilingDetection.test_nncf_validate_fq[Custom_Object_Detection_YOLOX] __________________________________________________________________________________________________

self = <tests.e2e.cli.detection.test_tiling_detection.TestToolsTilingDetection object at 0x7f2884c30f40>
template = ModelTemplate(model_template_id='Custom_Object_Detection_YOLOX', model_template_path='/home/yuchunli/training_extensio...paths=ExportableCodePaths(default=None, openvino=None), task_type_sort_priority=-1, gigaflops=6.5, size=20.4, hpo=None)
tmp_dir_path = PosixPath('/tmp/tmp278z3arm/tiling_det')

    @e2e_pytest_component
    @pytest.mark.skipif(TT_STABILITY_TESTS, reason="This is TT_STABILITY_TESTS")
    @pytest.mark.parametrize("template", templates, ids=templates_ids)
    def test_nncf_validate_fq(self, template, tmp_dir_path):
        tmp_dir_path = tmp_dir_path / "tiling_det"
        if template.entrypoints.nncf is None:
            pytest.skip("nncf entrypoint is none")
    
>       nncf_validate_fq_testing(template, tmp_dir_path, otx_dir, "detection", type(self).__name__)

tests/e2e/cli/detection/test_tiling_detection.py:219: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/test_suite/run_test_command.py:559: in nncf_validate_fq_testing
    _validate_fq_in_xml(xml_path, path_to_ref_data, "nncf", test_name)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

xml_path = '/tmp/tmp278z3arm/tiling_det/otx/algorithms/detection/configs/detection/cspdarknet_yolox/exported_nncf_Custom_Object_Detection_YOLOX/openvino.xml'
path_to_ref_data = '/home/yuchunli/training_extensions/tests/e2e/cli/detection/reference/Custom_Object_Detection_YOLOX/compressed_model.yml', compression_type = 'nncf', test_name = 'TestToolsTilingDetection'

    def _validate_fq_in_xml(xml_path, path_to_ref_data, compression_type, test_name):
        num_fq = get_number_of_fakequantizers_in_xml(xml_path)
        assert os.path.exists(path_to_ref_data), f"Reference file does not exist: {path_to_ref_data} [num_fq = {num_fq}]"
    
        with open(path_to_ref_data, encoding="utf-8") as stream:
            ref_data = yaml.safe_load(stream)
        ref_num_fq = ref_data.get(test_name, {}).get(compression_type, {}).get("number_of_fakequantizers", -1)
>       assert num_fq == ref_num_fq, f"Incorrect number of FQs in optimized model: {num_fq} != {ref_num_fq}"
E       AssertionError: Incorrect number of FQs in optimized model: 84 != -1

tests/test_suite/run_test_command.py:475: AssertionError
================================================================================================================================ warnings summary ================================================================================================================================
venv/otx/lib/python3.10/site-packages/_pytest/config/__init__.py:747
  /home/yuchunli/training_extensions/venv/otx/lib/python3.10/site-packages/_pytest/config/__init__.py:747: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: tests.test_suite.fixtures
    self.import_plugin(import_spec)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================================================================ short test summary info =============================================================================================================================
FAILED tests/e2e/cli/detection/test_tiling_detection.py::TestToolsTilingDetection::test_nncf_validate_fq[Custom_Object_Detection_Gen3_ATSS] - AssertionError: Incorrect number of FQs in optimized model: 155 != -1
FAILED tests/e2e/cli/detection/test_tiling_detection.py::TestToolsTilingDetection::test_nncf_validate_fq[Custom_Object_Detection_Gen3_SSD] - AssertionError: Incorrect number of FQs in optimized model: 67 != -1
FAILED tests/e2e/cli/detection/test_tiling_detection.py::TestToolsTilingDetection::test_nncf_validate_fq[Custom_Object_Detection_YOLOX] - AssertionError: Incorrect number of FQs in optimized model: 84 != -1
======================================================================================================== 3 failed, 60 passed, 9 skipped, 1 warning in 1353.00s (0:22:32) =========================================================================================================
```

**Review by cih9088 — DISMISSED** [review]

**Review by jaegukhyun — CHANGES_REQUESTED** [review]
Could you check e2e tiling instance segmentation e2e test too? And please change nncf fq number in reference.

**Review by sungmanc — CHANGES_REQUESTED** [review]
We could remove useless annotation files.
(image_info***.json, stuff_***.json)

**Review by jaegukhyun — DISMISSED** [review]

**Review by jaegukhyun — APPROVED** [review]

**Review by goodsong81 — APPROVED** [review]

**Comment by eugene123tw** [comment]
@jaegukhyun e2e result from test_tiling_instseg
```
tests/e2e/cli/instance_segmentation/test_tiling_instseg.py::TestToolsTilingInstanceSegmentation::test_pot_eval[Custom_Counting_Instance_Segmentation_MaskRCNN_ResNet50] [*] Update data configuration file to: /tmp/tmpjixxlccv/tiling_ins_seg/otx/algorithms/detection/configs/instance_segmentation/resnet50_maskrcnn/data.yaml
2023-05-03 00:12:05,661 | INFO : Loading OpenVINO OTXDetectionTask
2023-05-03 00:12:06,498 | INFO : OpenVINO task initialization completed
2023-05-03 00:12:06,499 | INFO : Start OpenVINO inference
2023-05-03 00:12:07,242 | INFO : 0.7425226770001245 secs
2023-05-03 00:12:07,911 | INFO : 0.6690832060003231 secs
2023-05-03 00:12:07,911 | INFO : Avg time per image: 0.7058029415002238 secs
2023-05-03 00:12:07,911 | INFO : Total time: 1.4116058830004476 secs
2023-05-03 00:12:07,911 | INFO : OpenVINO inference completed
2023-05-03 00:12:07,911 | INFO : Start OpenVINO metric evaluation
2023-05-03 00:12:07,918 | INFO : OpenVINO metric evaluation completed
Performance(score: 0.0, dashboard: (1 metric groups))
/home/yuchunli/training_extensions/venv/otx/lib/python3.10/site-packages/mmcv/__init__.py:20: UserWarning: On January 1, 2023, MMCV will release v2.0.0, in which it will remove components related to the training process and add a data transformation module. In addition, it will rename the package names mmcv to mmcv-lite and mmcv-full to mmcv. See https://github.com/open-mmlab/mmcv/blob/master/docs/en/compatibility.md for more details.
  warnings.warn(
/home/yuchunli/training_extensions/venv/otx/lib/python3.10/site-packages/openvino/offline_transformations/__init__.py:10: FutureWarning: The module is private and following namespace `offline_transformations` will be removed in the future, use `openvino.runtime.passes` instead!
  warnings.warn(
PASSED

=============================== warnings summary ===============================
venv/otx/lib/python3.10/site-packages/_pytest/config/__init__.py:747
  /home/yuchunli/training_extensions/venv/otx/lib/python3.10/site-packages/_pytest/config/__init__.py:747: PytestAssertRewriteWarning: Module already imported so cannot be rewritten: tests.test_suite.fixtures
    self.import_plugin(import_spec)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============ 34 passed, 14 skipped, 1 warning in 4400.66s (1:13:20) ===========
```

**Pull request merged**